usaspending_api/etl/tests/integration/test_load_transactions_in_delta.py

```diff
@@ -804,8 +803,436 @@ def test_inserts_and_deletes(
+    def test_initial_load_no_transaction_id_lookup(
+        self, spark, s3_unittest_data_bucket, hive_unittest_metastore_db, populate_initial_postgres_data
+    ):
+        TestTransactionFabs.initial_load_no_transaction_id_lookup(spark, s3_unittest_data_bucket)
+
+        # Verify key fields in transaction_fabs table
+        query = (
+            f"SELECT {', '.join(TestTransactionFabs.transaction_fabs_compare_fields)} FROM int.transaction_fabs "
+            "ORDER BY published_fabs_id"
+        )
         delta_data = [row.asDict() for row in spark.sql(query).collect()]
-        assert equal_datasets(expected_award_id_lookup, delta_data, "")
+
+        # Even without the call to load_transactions_in_delta with etl-level of transaction_id_lookup, the appropriate
+        # data will be populated in the transaction_id_lookup table via initial_run to allow the call to
+        # load_transactions_in_delta with etl-level of transaction_fabs to populate int.transaction_fabs correctly.
+        assert equal_datasets(TestTransactionFabs.expected_initial_transaction_fabs, delta_data, "")
+
+    @mark.django_db(transaction=True)
+    def test_inserts_updates_deletes_no_transaction_id_lookup(
+        self, spark, s3_unittest_data_bucket, hive_unittest_metastore_db, populate_initial_postgres_data
+    ):
+        # Even without the call to load_transactions_in_delta with etl-level of transaction_id_lookup, the appropriate
+        # data will be populated in the transaction_id_lookup table via initial_run to allow the call to
+        # load_transactions_in_delta with etl-level of transaction_fabs to populate int.transaction_fabs correctly with
+        # the initial data.
+        TestTransactionFabs.initial_load_no_transaction_id_lookup(spark, s3_unittest_data_bucket)
+
+        # Can't use spark.sql to just insert rows with only values for desired columns (need to specify values for
+        # all of them), so using model baker to add new rows to Postgres table, and then pushing new table to Delta.
+        insert_update_datetime = datetime.now(timezone.utc)
+        baker.make(
+            "transactions.SourceAssistanceTransaction",
+            published_fabs_id=6,
+            afa_generated_unique="award_assist_0004_trans_0001",
+            updated_at=insert_update_datetime,
+            is_active=True,
+            unique_award_key="award_assist_0004",
+        )
+        load_delta_table_from_postgres("published_fabs", s3_unittest_data_bucket)
+
+        spark.sql(
+            f"""
+            UPDATE raw.published_fabs
+            SET updated_at = '{insert_update_datetime}'
+            WHERE published_fabs_id = 4 OR published_fabs_id = 5
+        """
+        )
+
+        spark.sql(
+            f"""
+            DELETE FROM raw.published_fabs
+            WHERE published_fabs_id = 2 OR published_fabs_id = 3
+        """
+        )
+
+        call_command("load_transactions_in_delta", "--etl-level", "transaction_fabs")
```

Comment:
Shouldn't an extra call to `load_transactions_in_delta` with an `--etl-level` of `transaction_id_lookup` be necessary here to capture the newly inserted transaction?

Reply:
I was specifically testing the behavior if no call to `load_transactions_in_delta` with an `--etl-level` of `transaction_id_lookup` was made.  It's actually more important for this test that no such call is made there, as any such call immediately after calling `load_transactions_in_delta` with an `--etl-level` of `initial_run` won't actually do anything.